mlir/lib/Dialect/EmitC/IR/EmitC.cpp

```diff
@@ -116,6 +116,7 @@ bool mlir::emitc::isIntegerIndexOrOpaqueType(Type type) {
 bool mlir::emitc::isSupportedFloatType(Type type) {
   if (auto floatType = llvm::dyn_cast<FloatType>(type)) {
     switch (floatType.getWidth()) {
+    case 16:
```

Comment:
Could you please explicitly check for Float16Type and BFloat16Type here?

Reply:
Done